benchmarking/README.md

```diff
@@ -0,0 +1,568 @@
+  - name: gdrive
+    enabled: false
+    drive_folder_id: ${GDRIVE_FOLDER_ID}
+    service_account_file: ${GDRIVE_SERVICE_ACCOUNT_FILE}
+
+# Optional: Define datasets for template substitution
+datasets:
+  - name: common_crawl
+    formats:
+      - type: json
+        path: "{datasets_path}/cc_sample.jsonl"  # Can reference base paths
+      - type: parquet
+        path: "{datasets_path}/cc_sample.parquet"
+
+# Required: List of benchmark entries to run
+entries:
+  - name: my_benchmark
+    enabled: true  # Optional: Whether to run this entry (default: true)
+    script: my_script.py
+    args: >-
+      --input {dataset:common_crawl,parquet}
+      --output {session_entry_dir}/output
+    timeout_s: 1800  # Optional: Override global timeout
+
+    # Optional: Per-entry sink configuration
+    sink_data:
+      - name: slack
+        additional_metrics: ["throughput_docs_per_sec", "num_documents_processed"]
+
+    # Optional: Ray configuration for this entry
+    ray:
+      num_cpus: 32
+      num_gpus: 1
+      enable_object_spilling: false
+
+    # Optional: Requirements for the benchmark to pass
+    requirements:
+      - metric: throughput_docs_per_sec
+        min_value: 100
+
+    # Optional: Override global delete_scratch setting
+    delete_scratch: false
+```
+
+### Passing Configuration Files
+
+**Multiple config files:**
+
+```bash
+python benchmarking/run.py \
+  --config config.yaml \
+  --config paths.yaml \
+  --config machine_specific.yaml
+```
+
+Files are merged in order. Later files override earlier ones for conflicting keys.
+
+**Session naming:**
+
+```bash
+python benchmarking/run.py \
+  --config config.yaml \
+  --session-name my-experiment-v2
+```
+
+### Environment Variables
+
+Configuration values can reference environment variables using `${VAR_NAME}` syntax:
+
+```yaml
+results_path: "${HOME}/benchmarks/results"
+sinks:
+  - name: slack
+    webhook_url: ${SLACK_WEBHOOK_URL}
+  - name: mlflow
+    tracking_uri: ${MLFLOW_TRACKING_URI}
+```
+
+### Template Substitution and Path Resolution
+
+The framework supports several types of placeholders in configuration values:
+
+**Base path references** - Reference the configured base paths:
+
+```yaml
+datasets:
+  - name: my_dataset
+    formats:
+      - type: parquet
+        path: "{datasets_path}/subdir/data.parquet"
+```
+
+Available base path placeholders:
+- `{results_path}` - Resolves to the configured `results_path`
+- `{artifacts_path}` - Resolves to the configured `artifacts_path`
+- `{datasets_path}` - Resolves to the configured `datasets_path`
+
+**Dataset references** - Reference datasets in entry arguments:
+
+```yaml
+args: --input {dataset:common_crawl,parquet}
+```
+
+Resolves to the path defined in the `datasets` section for that dataset and format.
+
+**Session entry directory** - Reference the entry's runtime directory:
+
+```yaml
+args: --output {session_entry_dir}/results
+```
```

Comment:
after reading this i'm slightly confused with where `artifacts_path` is used? is the base of`session_entry_dir`, `artifacts_path`?
what does `delete_scratch : True` do? can we elaborate on those?